test/interpreter_scripts/errors/namespace_reassign_const_nested.txt

```diff
@@ -0,0 +1,11 @@
+namespace outer {
+    namespace inner {
+        y = 0
+    }
+}
+
+namespace outer {
+    namespace inner { //TODO: fix resolution
+        y = 1
+    }
+}
```

Comment:
The ability to re-enter named scopes needs thought. The single-lookup resolution of lexical scopes is a thing of beauty, but won't solve the named scope re-entry problem in its current form.

Reply:
Honestly, I would tend to add scope methods to the map when you enter, and remove when you leave.

Cons:
- You need to think about efficiency of retrieving scope variables.
- Poor performance when rapidly entering and leaving scopes, e.g. an implementation file with lots of MyClass:: method
- Still need to handle resolution to parent scopes

Pros:
- Easy to implement
- Single-lookup read

At least that's a working implementation, and you can figure out something better at your leisure